From 9f91df2afad50be5d0c5aa42c1fe1b86d1a27f90 Mon Sep 17 00:00:00 2001
From: AsavarTzeth <asavartzeth@gmail.com>
Date: Sun, 17 Feb 2019 11:09:56 +0100
Subject: [PATCH 2/3] Disable Advanced SIMD or NEON on arm

This might work if compiled on actual arm hardware, that supports this
feature. However test builds won't pass with this enabled and I assume
flathub buildbot builds will fail as well.

These are the errors that occur:
{standard input}: Assembler messages:
{standard input}:15039: Error: garbage following instruction -- `orr r5,r5,LSL $1'
{standard input}:15207: Error: garbage following instruction -- `orr r8,r8,LSL $1'
{standard input}:15522: Error: garbage following instruction -- `orr r5,r5,LSL $1'
{standard input}:15688: Error: garbage following instruction -- `orr lr,lr,LSL $1'
  g++ decode.cpp
  g++ phase_detector.cpp
  g++ widgets/LSPButton.cpp
  g++ CtlSwitch.cpp
  g++ CtlValuePort.cpp
{standard input}:19681: Error: garbage following instruction -- `subs r5,r3,LSL $1'
{standard input}:20600: Error: garbage following instruction -- `subs r6,r3,LSL $1'
  g++ graph_equalizer.cpp
make[2]: *** [Makefile:77: /run/build/lsp-plugins/.build/src/dsp/neon-d32.o] Error 1
make[1]: *** [Makefile:57: dsp] Error 2
---
 src/dsp/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dsp/Makefile b/src/dsp/Makefile
index 97ed47f3..b2a4e678 100644
--- a/src/dsp/Makefile
+++ b/src/dsp/Makefile
@@ -31,7 +31,7 @@ ifeq ($(BUILD_PROFILE), x86_64)
 LINK_OBJECTS           += $(X86_IMPL) $(SSE_IMPL) $(SSE2_IMPL) $(SSE3_IMPL) $(SSE4_IMPL) $(AVX_IMPL) $(AVX2_IMPL)
 endif
 ifeq ($(BUILD_PROFILE), armv7a)
-LINK_OBJECTS           += $(ARM_IMPL) $(NEON_D32_IMPL)
+#LINK_OBJECTS           += $(ARM_IMPL) $(NEON_D32_IMPL)
 endif
 
 .PHONY: all target
-- 
2.20.1

